### DISCUSSION TITLE
Multiple acme2certifiers for redundancy

### DISCUSSION THREAD

**Discussion by Flo451** [grindsa/acme2certifier#164] [Q&A] [answered]
Hi,

I am exploring how acme2certifier supports running in either an active/active or load-balanced fashion. I have previously encountered small outages where either the server or the backend database became inaccessible. This has led me to wonder about feasible designs that would allow running separate instances that can be switched between seamlessly.

I assume that there is some critical state, such as account registrations and previous orders, that needs to be shared between backends to achieve this setup.

I am curious to hear your thoughts on this.

**Comment by grindsa** [comment]
Hi,

Running multiple acme2certifier instance in an active/active setup should work in principle; I see no reasons why it should not work.

Some first thoughts:

1. You need to use the django_db_handler and an active/active database like postgres with multi-master replication or mariadb with Galera; the default wsgi_handler is limited to SQLite which is not designed for concurrent write operations
2. You need to shared storage for the config-file (acme_srv.cfg) and some ca_handler related files and directories

If you have a recommendation for #2 I can go for a small POC and create some documentation around it (I have this on my list for quite some time)

**Comment by grindsa** [comment, marked as answer]
Hi again,

i wrote a [small tutorial](https://github.com/grindsa/acme2certifier/blob/devel/docs/a2c-ubuntu-loadbalancing.md) how to setup  two a2c service in a load-balancing mode.  Feel free to test. Feedback is always welcome...
